db/migrate/20170517222310_change_annotation_author_to_not_null.rb

```diff
@@ -2,6 +2,7 @@ class ChangeAnnotationAuthorToNotNull < ActiveRecord::Migration[5.1]
   def do_sql(*args)
     expression = ActiveRecord::Base.send :sanitize_sql, args
     ActiveRecord::Base.connection.exec_query(expression)
+    raise StandardError, 'You can’t run this migration without any users in your DB'
```

Comment:
This will always raise, which is definitely a problem. I was thinking in `up`, you might use `.first.try(:[], 'id')` to get the user ID, then raise the error if it’s `nil`. e.g:

```rb
def up
  default_author_id = do_sql(
    'SELECT id FROM users ORDER BY created_at ASC LIMIT 1'
  ).first.try(:[], 'id')

  if default_author_id.nil?
    raise StandardError, 'You can’t run this migration without any users in your DB'
  end
```

Reply:
Hmm, maybe simpler and better to not use `try` and just separate the steps of getting the user and the user’s ID:

```rb
default_author = do_sql(
  'SELECT id FROM users ORDER BY created_at ASC LIMIT 1'
).first

if default_author.nil?
  raise StandardError, 'You can’t run this migration without any users in your DB'
end

do_sql(
  'UPDATE annotations SET author_id = ? WHERE author_id IS NULL',
  default_author['id']
)
```